vumi_message_store/tests/test_riak_backend.py

```diff
@@ -1205,3 +1221,52 @@ def test_list_message_event_keys_with_statuses_no_message(self):
         keys_page = yield self.backend.list_message_event_keys_with_statuses(
             "badmsg")
         self.assertEqual(list(keys_page), [])
+
+
+class TestMessageStoreRiakBackend(RiakBackendTestMixin, VumiTestCase):
+
+    def setUp(self):
+        self.persistence_helper = self.add_helper(
+            PersistenceHelper(use_riak=True))
+        self.manager = self.persistence_helper.get_riak_manager()
+        self.set_up_tests()
+
+    def get_backend(self):
+        return MessageStoreRiakBackend(self.manager)
+
+
+class TestMessageStoreRiakBackendInMemory(RiakBackendTestMixin, VumiTestCase):
+
+    def setUp(self):
+        self.state = FakeRiakState(is_sync=False)
+        self.add_cleanup(self.state.teardown)
+        self.manager = FakeMemoryRiakManager(self.state)
+        self.set_up_tests()
+
+    def get_backend(self):
+        return MessageStoreRiakBackend(self.manager)
+
+
+class TestMessageStoreRiakBackendSync(RiakBackendTestMixin, VumiTestCase):
+
+    def setUp(self):
+        self.persistence_helper = self.add_helper(
+            PersistenceHelper(use_riak=True, is_sync=True))
+        self.manager = self.persistence_helper.get_riak_manager()
+        self.set_up_tests()
+
+    def get_backend(self):
+        return MessageStoreRiakBackend(self.manager)
+
+
+class TestMessageStoreRiakBackendInMemorySync(RiakBackendTestMixin,
+                                              VumiTestCase):
+
+    def setUp(self):
+        self.state = FakeRiakState(is_sync=True)
+        self.add_cleanup(self.state.teardown)
+        self.manager = FakeMemoryRiakManager(self.state)
+        self.set_up_tests()
+
+    def get_backend(self):
+        return MessageStoreRiakBackend(self.manager)
```

Comment:
They were different in an earlier implementation attempt. I'll move that up into the base class and kill the duplication.